staging/src/k8s.io/code-generator/cmd/generic-conversion-gen/generators/conversion_generator.go

```diff
@@ -0,0 +1,797 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+	"k8s.io/klog"
+)
+
+// ConversionGenerator is a generic, configurable gengo generator to generate
```

Comment:
I think this is too much to copy; can we factor out the specific bits into some sort of policy interface which the two generators can each implement?

(if this is a move I must be missing the corresponding deletion, sorry)

Reply:
Ah, I think the next commit removes significant pieces of the existing copy. Hm.